### PR TITLE
Workflow handling

### PR DESCRIPTION
**Pull request by joewallwork**
# Workflow handling

In the interests of green computing, development efficiency and avoiding unnecessary CI runs, this PR sets up paths for workflow handling, so that the linters are only run if the source code or build code change and the test suite is only run if the test suite, build system, or config files change.

The PR also fixes a copy-paste error in the linting workflow (introduced in #759).

### CODE REVIEW

**Comment by joewallwork** [comment]
If you check the Actions logs, you should see that:
* My initial push (at https://github.com/nextsimhub/nextsimdg/pull/760/commits/9d1059e21087960b6340e80a5c8e96b0822e0954) didn't trigger any CI jobs because nothing in the source code, build system, or config files changed.
* My second push (at https://github.com/nextsimhub/nextsimdg/pull/760/commits/61523e8025fe14c0832e32fdcae29f044027c9d1) triggered the linting workflows (but not test workflows) because I added that they should be run if the `linting.yml` workflow file itself changes.
* My third push (at https://github.com/nextsimhub/nextsimdg/pull/760/commits/562bd6dddb5212f0f4d62fbc7be7f40773cdc78f) also triggered the test suite workflows because I added that they should be run if the `test_suite.yml` workflow file itself changes.

File: .github/workflows/linting.yml [hunk 1]
```diff
@@ -5,13 +5,18 @@ on:
...
+      - '**.hpp'
+      - '**CMakeLists.txt'
 
 jobs:
   clang-format:
     runs-on: ubuntu-24.04
     steps:
     - uses: actions/checkout@v2
-    - name: clang-format
+    - name: cmake-format
```
Comment (einola) [inline .github/workflows/linting.yml]: Doesn't this change belong in line 29/34?

**Review by einola — CHANGES_REQUESTED** [review]
Sounds like a good idea!

But isn't there a mistake on line 14/19 cf line 29/34 in linting.yml?

File: .github/workflows/linting.yml (hunk 1 above)
Comment (joewallwork) [reply .github/workflows/linting.yml]: Oops! Good spot - fixed in 470d3bd9.

**Review by einola — APPROVED** [review]

**Pull request merged**